### PR TITLE
remove activated feature set_exempt_rent_epoch_max

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
feature #28683 was activated on all clusters.

#### Summary of Changes
Remove code for handling prior to feature activation.

Fixes #

### CODE REVIEW

File: runtime/src/bank/tests.rs [hunk 1]
```diff
@@ -1651,9 +1647,8 @@ fn test_rent_eager_collect_rent_in_partition(should_collect_rent: bool) {
 
     let (mut genesis_config, _mint_keypair) = create_genesis_config(1_000_000);
     for feature_id in FeatureSet::default().inactive {
-        if feature_id != solana_sdk::feature_set::set_exempt_rent_epoch_max::id()
-            && (!should_collect_rent
-                || feature_id != solana_sdk::feature_set::disable_rent_fees_collection::id())
+        if !should_collect_rent
```
[the remark this answers is not in the document]
Comment (HaoranYi) [reply runtime/src/bank/tests.rs]: The fix [https://github.com/solana-labs/solana/pull/35082/commits/d214e787263e5674e6bbe69ecc230dacebc3c27d] looks good to me.

File: runtime/src/bank/tests.rs [hunk 2]
```diff
@@ -11896,87 +11854,6 @@ fn test_feature_hashes_per_tick() {
     assert_eq!(bank.hashes_per_tick, Some(UPDATED_HASHES_PER_TICK6));
 }
 
-#[test_case(true)]
-#[test_case(false)]
-fn test_stake_account_consistency_with_rent_epoch_max_feature(
-    rent_epoch_max_enabled_initially: bool,
-) {
-    // this test can be removed once set_exempt_rent_epoch_max gets activated
```
Comment (jeffwashington) [inline runtime/src/bank/tests.rs:11904]: a nice note for the future (ie. now!)

**Review by HaoranYi — APPROVED** [review]
lgtm!

**Comment by HaoranYi** [comment]
I rebased to master to resolve the conflicts.

**Pull request merged**